extensions/amp-delight-player/0.1/amp-delight-player.js

```diff
@@ -0,0 +1,574 @@
+    this.playedRanges_ = [];
+
+    /** @private {boolean} */
+    this.isFullscreen_ = false;
+
+    /** @private {Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?Function} */
+    this.unlistenScreenOrientationChange_ = null;
+
+    /** @private {?Function} */
+    this.unlistenOrientationChange_ = null;
+
+    /** @private {?Function} */
+    this.unlistenDeviceOrientation_ = null;
+
+    /** @private {?Function} */
+    this.unlistenDeviceMotion_ = null;
+
+    /** @private {HTMLElement} */
+    this.placeholderEl_ = null;
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(this.baseURL_, onLayout);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.contentID_ = user().assert(
+        this.element.getAttribute('data-content-id'),
+        'The data-content-id attribute is required'
+    );
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const src = `${this.baseURL_}/player/${this.contentID_}?amp=1`;
+    const iframe = createFrameFor(this, src);
+
+    iframe.setAttribute('allow', 'vr');
+
+    this.unlistenMessage_ = listen(this.win, 'message', event => {
+      this.handleDelightMessage_(event);
+    });
+
+    this.iframe_ = iframe;
+
+    this.registerEventHandlers_();
+
+    return this.loadPromise(iframe);
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    this.unregisterEventHandlers_();
+
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    const src = `${this.baseURL_}/poster/${this.contentID_}`;
+    placeholder.setAttribute('placeholder', '');
+
+    setStyle(placeholder, 'background-image', `url(${src})`);
+
+    this.placeholderEl_ = placeholder;
+
+    return placeholder;
+  }
+
+  /** @override */
+  firstLayoutCompleted() {
+    const el = this.placeholderEl_;
+    let promise = null;
+    if (el && this.isInViewport()) {
+      el.classList.add('i-amphtml-faded');
```

Comment:
`i-amphtml-faded` is too generic for a component. Please use `i-amphtml-delight-player-faded`